src/lib/__tests__/InstantSearch-test.tsx

```diff
@@ -1540,8 +1540,8 @@ describe('use', () => {
       .invocationCallOrder[0];
     const middlewareSubscribeCallOrder =
       middlewareSpy.subscribe.mock.invocationCallOrder[0];
-    // Checks that `mainIndex.init` was called before subscribing the middleware.
-    expect(widgetsInitCallOrder).toBeLessThan(middlewareSubscribeCallOrder);
+    // Checks that `mainIndex.init` was called after subscribing the middleware.
+    expect(widgetsInitCallOrder).toBeGreaterThan(middlewareSubscribeCallOrder);
```

Comment:
This is technically a breaking change, but I don't think it will break anything in real life.

Reply:
We've got a customer report saying that their app broke. I think that's because of this commit, because `instantSearchInstance.renderState[instantSearchInstance.indexName]` is undefined in `createInsightsMiddleware`.

Any middleware that relies on `instantSearchInstance` might break now, because there's no certainty that the instance has started anymore.

Could we rather have a `setInitialUiState` method which updates `initialUiState` without these PR changes?

Reply:
We actually fixed that regression at https://github.com/algolia/instantsearch.js/pull/4897

I tried to fix the issue without introducing new API, but I let you and the team decide whether to revert #4849 and #4897 and introduce `setInitialUiState`.